### ISSUE TITLE
Modify Armor Values and Upgrade Values

### ISSUE THREAD

**Issue by Pinkhai** [iron431/Irons-Spells-n-Spellbooks#310]
### Feature description

Players are able to modify how much power 1 orb could give. ex: 1 upgrade_orb could give 2.5% spellpower. After that, players could modify it to 5% or more in the config file
Make armors compatible to kubejs or something else. I found a fact that armors are unable to adjust in kubejs . I have asked that in Irons_spellbooks discord, someone told me he has the same issue, he doubted that armors were made by special methods. could you please make armors be custom?

### How it improves the player experience

It would be much more convenient  for modpack authers. At the same time, the mod's number are adapt to vanilla, it would become not powerful enough when combine with some mods which added violent mobs(like the Apotheosis boss), players may need a way to strengthen magic

**Comment by iron431** [comment]
I've never used kubejs but you can definitely use crafttweaker to edit the armors. Customizable upgrades are already planned.

**Comment by iron431** [comment]
configurable armor is also planned but not a high priority

**Comment by Auuki** [comment]
+1
I'd like to rebalance some armors, for Instance Wandering set to match Leather set in terms of armor values. Currently I can't do so neither through datapack nor config or even kubeJS as mentioned by Pinkhai. I found that odd considering how nicely we can configure spells. 1.20.1 version of the mod.

**Comment by iron431** [comment]
attribute rebalance can be done by numerous mods, including kubejs or crafttweaker

**Comment by Auuki** [comment]
> attribute rebalance can be done by numerous mods, including kubejs or crafttweaker

Here I'm using kubeJS' standard method for such task:
```
ItemEvents.modification(event => {
  event.modify('irons_spellbooks:wandering_magician_boots', item => {
    item.armorProtection = 1.0
  })
  event.modify('minecraft:leather_boots', item => {
    item.armorProtection = 11.0
  })
})
```
Tested both in my modpack I'm working on as well as a bare modpack in case it's caused by some other mod. Wandering Magician Shoes are untouched, Leather Boots were correctly changed. Found some users having similar problem on KubeJS discord. As for CraftTweaker, I'm not aware of a method that would allow me to do that in 1.20.1 or any recent version. I think the last time I saw it was in CraftTweaker around MC 1.12 and since then only in some addon mods that come and go. Property Modifier was also an option but it causes incompatibility issues on 1.20.1.

**Comment by iron431** [comment]
https://docs.blamejared.com/1.20.1/en/vanilla/api/item/IItemStack#addGlobalAttributeModifier

**Comment by Auuki** [comment]
> https://docs.blamejared.com/1.20.1/en/vanilla/api/item/IItemStack#addGlobalAttributeModifier

Right, I forgot about this one as this is technically also possible in KubeJS, although the code looks worse. The difference in this approach is that it does exactly what the function is called - adds a global attribute modifier, while presented KubeJS solution actually changes/overwrites the value (latter looks normal in the expanded tooltip). I guess I can work with that.

**Comment by iron431** [comment]
upgrade orbs are datadriven. configurable armor is not necessarily planned
